airflow/api_connexion/endpoints/connection_endpoint.py

```diff
@@ -47,31 +55,53 @@ def get_connection(connection_id, session):
 
 
 @provide_session
-def get_connections(session):
+def get_connections(session, limit, offset=0):
     """
     Get all connection entries
     """
-    offset = request.args.get(parameters.page_offset, 0)
-    limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
-
+    total_entries = session.query(func.count(Connection.id)).scalar()
     query = session.query(Connection)
-    total_entries = query.count()
-    query = query.offset(offset).limit(limit)
-
-    connections = query.all()
+    connections = query.offset(offset).limit(limit).all()
     return connection_collection_schema.dump(ConnectionCollection(connections=connections,
                                                                   total_entries=total_entries))
 
 
-def patch_connection():
+@provide_session
+def patch_connection(connection_id, session, update_mask=None):
     """
     Update a connection entry
     """
-    raise NotImplementedError("Not implemented yet.")
+    body = request.json
+    connection = session.query(Connection).filter_by(conn_id=connection_id).first()
+    if connection is None:
+        raise NotFound("Connection not found")
+    if update_mask:
```

Comment:
Can you also add validation for the mask? The mask should not contain fields that are on the allowed list. In particular, you should not be able to save any data to the _password field. This is a very common mistake, which leads to CATastrophe.

Reply:
Hi, check if this is ok https://github.com/apache/airflow/commit/09a7ab8be098059f1a388eeebd09610169a83117

Reply:
Ah. Probably I was wrong, but I will still have to check how connexion behaves when it sees additional fields in the request from the user. It seems to me that it could be set in the connexion configuration, but I'm not sure. I will come back to this because it will take me a moment to check.

Reply:
Also check this, https://github.com/marshmallow-code/marshmallow-sqlalchemy/issues/33#issuecomment-147008000

 For updatable fields, looks like we would need it in dagruns update